qiskit/backends/ibmq/ibmqbackend.py

```diff
@@ -46,33 +47,46 @@ def __init__(self, configuration, api=None):
         """
         super().__init__(configuration=configuration)
         self._api = api
-
+        self._submit_info = None   # dictionary coming from api upon submission
         if self._configuration:
             configuration_edit = {}
             for key, vals in self._configuration.items():
                 new_key = _snake_case_to_camel_case(key)
                 configuration_edit[new_key] = vals
+            #  FIXME: This is a hack as the hpc simulator is not correct in api
+            if configuration_edit['name'] == 'ibmqx_hpc_qasm_simulator':
+                configuration_edit['simulator'] = True
             self._configuration = configuration_edit
             # FIXME: This is a hack to make sure that the
             # local : False is added to the online device
             self._configuration['local'] = False
 
     def run(self, q_job):
```

Comment:
can we also define a `jobs` (or `get_jobs`) method for the ibmqbackend? this would return a list of job ids ever executed on this backend, from this user. a lot of people have requested this.

Under the hood, i think we can just do `api.get_jobs(limit=5000)`, and then filter to keep only those jobs that have run on this backend. We can ask for API improvements later that does the filtering for us.

Reply:
So I was going to implement this `ibmqbackend.jobs()` functionality, but it was intended to return IBMQJob instances, which is more convenient as you only have one consistent interface to deal with Jobs. The problem is that we can't create these instances because there's a dependency between some qobj data fields that are not being stored in the API and the Result component, so we still miss some of the bits. My plan was going to review these dependencies because at first glance I saw that they might not be needed, so by removing them we could gain the ability to instance IBMQJobs and reduce a lot of complexity at the same time.

Reply:
@atilag i agree that reconstructing the full object requires more thought. But for the near term can we have a convenience function, maybe in the wrapper, so users can get their past results in dictionary format? A code like this already works, we just need to put it in the SDK.

```
from IBMQuantumExperience import IBMQuantumExperience as IBMQE

def get_jobs(APItoken, config, limit=50, completed=True):
    """
    Return results of IBM-Q Experience jobs.
    
    Params:
        APItoken (str): todo
        config (dict): todo
        limit (int): todo
        completed (bool): todo
        
    Returns:
        A list of job results
    """
    ibmq = IBMQE(APItoken, config)
    my_jobs = ibmq.get_jobs(limit=limit)

    #filter down to get a list of completed jobs
    if completed:
        return [j for j in my_jobs if j['status']=='COMPLETED']
    else:
        return my_jobs
```